### DISCUSSION TITLE
Support VEX2 files?

### DISCUSSION THREAD

**Discussion by vlbi-jun-yang** [nvi-inc/fs#188] [Q&A] [answered]
During the recent K-band experiment N22K1, Onsala got its DBBC3 backend to participate in the NME observations in parallel. I have learned that the JIVE SFXC required a VEX2 file to correlate multi-thread multi-channel VDIF data.  I have looked at our local FS `drudg`. Currently, it only support VEX1.5 files.  So, it would be great for FS drudg to also support VEX2 files in the near future. 

BTW, [the recent NRAO Sched 11.7](http://www.aoc.nrao.edu/software/sched/#x1-3100005.1) has started to support VEX2 to a certain degree.

**Comment by wehimwich** [comment, marked as answer]
Supporting VEX2 is certainly in the plan. It _might_ be up next after supporting FSL11 and expanding the length of length of _.snp_ (and _.prc_ and _.log_) file names to support the new, longer, IVS session codes. Both are under development and hopefully will be ready soon.

My understanding is that during the transition to VEX2, VEX1 schedules will continue to be produced until all components can switch to VEX2. _drudg_ may be the last piece. There will probably need to be a significant amount of testing before the use of VEX1 can end altogether.

**Reply by vlbi-jun-yang** [reply]
Thanks for the feedback. I have just read [the online definition of VEX2](https://safe.nrao.edu/wiki/bin/view/VLBA/Vex2doc#A_61_36DAS_61_Block_40Substantially_changed_for_VEX2_41). It does have a great number of new features.  For example. 

> VEX2 is significantly different from VEX1 in that it allows much more flexibility in the number of back-ends and recorders and how they are interconnected. While VEX2 allows virtually any details of the configuration and wiring from the IF to the recording to be described, it is expected that this flexibility will only be used to specify necessary details of the systems that are not normally static. Few details need to be specified for most systems. In fact it is possible that systems are described adequately by a "virtual" configuration with no details of the physical system in use. What needs to be described is system specific.

It sounds really cool.
